### PR TITLE
Schema-attribute-to-mapped-attribute matching for 'Schemas' endpoint

### PR DESCRIPTION
**Pull request by pond**
Address #119 with opt-in configuration.

Reviewing by starting with the diff of `config/initializers/scimitar.rb` is recommended, as this explains the new configuration option and, with that, the controller code which uses it will obviously make more sense.

Hiding white space changes will reduce diff noise in the tests, where a chunk of code got indented into a new `context`.

### CODE REVIEW

File: spec/models/scimitar/resources/base_spec.rb [hunk 1]
```diff
@@ -4,7 +4,7 @@
   context 'basic operation' do
     FirstCustomSchema = Class.new(Scimitar::Schema::Base) do
       def self.id
-        'custom-id'
+        'urn:ietf:params:scim:schemas:custom-id'
```
Comment (pond) [inline spec/models/scimitar/resources/base_spec.rb:7]: A bunch of schema ID drive-bys given recent recommendations to be mindful of the RFC IANA information re. URNs in https://datatracker.ietf.org/doc/html/rfc7643#section-10.

**Review by bagp1 — APPROVED** [review]
